lib/jsonapi/utils/support/filter.rb

```diff
@@ -17,11 +17,12 @@ def apply_filter?(records, options = {})
 
         def filter_params
           @_filter_params ||=
-            if params[:filter].is_a?(Hash)
-              params[:filter].keys.each_with_object({}) do |resource, hash|
-                hash[resource] = params[:filter][resource]
-              end
+          case params[:filter]
```

Comment:
In Rails 5 Action Controller Parameters doesn't inherit from Hash so the `.is_a` method will return false and the proper filter isn't returned.